flink-core/src/main/java/org/apache/flink/configuration/TaskManagerOptions.java

```diff
@@ -83,12 +85,26 @@
 	public static final ConfigOption<String> HOST =
 		key("taskmanager.host")
 			.noDefaultValue()
-			.withDescription("The hostname of the network interface that the TaskManager binds to. By default, the" +
-				" TaskManager searches for network interfaces that can connect to the JobManager and other TaskManagers." +
-				" This option can be used to define a hostname if that strategy fails for some reason. Because" +
-				" different TaskManagers need different values for this option, it usually is specified in an" +
+			.withDescription("The address of the network interface that the TaskManager binds to." +
+				" This option can be used to define explicitly a binding address. Because" +
+				" different TaskManagers need different values for this option, usually it is specified in an" +
 				" additional non-shared TaskManager-specific config file.");
 
+	/**
+	 * The config parameter for automatically defining the TaskManager's binding address,
+	 * if {@link #HOST} configuration option is not set.
+	 */
+	public static final ConfigOption<String> HOST_BIND_POLICY =
+		key("taskmanager.host.bind-policy")
```

Comment:
should we rename this option to `network.taskmanager.bind-policy`? Then we could later extend it to `network.bind-policy` and `network.jobmanager.bind-policy`.

Reply:
Done.

Reply:
Renamed to `taskmanager.network.bind-policy` to be consistent with other existing config options (`taskmanager.network.*`).

Later the potentially extended config options could be named as `jobmanager.network.bind-address`, `network.bind-address`.